### DISCUSSION TITLE
MQTT can not be configured, since changes in OpenDTU can not be saved and freeze ESP32

### DISCUSSION THREAD

**Discussion by StKgl** [tbnobody/OpenDTU#2167] [Q&A]
My installation with a HMS-2000-4T is running for almost two weeks, but since I try to connect with MQTT im am getting crazy. In console, you can alwas see "Disconnect reason:MQTT_NOT_AUTHORIZED". 

Meanwhile, I figured out that most of my changes are ignored by OpenDTU. I could change my Wifi credential, but not the values in MQTT panel. Also the frequency in the DTU-properties could not be changed nor the power values for each panel in the inverter tab. Additionally, my last attempt most likely crashed the ESP32 and locked it. I had to unplug power and repower it again. Interestingly, the counter of how many configurations have been saved, remains at 17. 

I soldered a small board with an ESP32-D0WD-V3, a CMT2300A and a small OLED, which seem to work so far. In my home assistant installation, MQTT works with several other devices.

What kind of logfiles, screenshots or additional information do you need?

**Comment by stefan123t** [comment]
This sounds like you are having the MQTT brokers credentials wrong. Can you configure your broker to require username and password and enter them in OpenDTU ?
You can always take a backup of your config.json and/or pin_mapping.json in case you prefer to start from scratch they may provide you valuable insights in how you configured it in the past (while it was still working). You may restore those settings too.

**Reply by StKgl** [reply]
My credentials were the first thing I checked several times, but they are valid at several other devices. When I tried to change them, I figured out that they are not saved at all in OpenDTU! In the mqtt-tab, there is no username saved and mqtt anonymous connection is intentionally disabled in my broker.

**Reply by avandorp** [reply]
Do you use TLS with MQTT? If so, try the newest release, the older ones could get stuck.

**Reply by StKgl** [reply]
Thanks for your reply! No, I am/was not using MQTT with TLS. 

Since the counter for configuration changes stopped counting, I decided to flash the ESP32 again with the latest firmware of OpenDTU. Now the integration in MQTT works as expected. I have no idea what happened, but OpenDTU was not able so save any configuration changes.
